src/sagas/combat-sagas.tsx

```diff
@@ -0,0 +1,193 @@
+  );
+
+  yield sendMessage(
+    `${enemyCombatant.name} chose to ${enemySelectedCombatAction.name}.`
+  );
+}
+
+function* enterCombatListener() {
+  yield takeEvery(ENTER_COMBAT, selectEnemyAction);
+}
+
+function* selectEnemyAction() {
+  const enemyCombatant = yield select((state: StoreState) =>
+    state ? state.enemyCombatant : enemyCombatantReducerDefaultState
+  );
+
+  const selectedMove = _.sample(enemyCombatant.moveList);
+  const selectEnemyMoveAction: SelectEnemyCombatActionAction = {
+    type: SELECT_ENEMY_COMBAT_ACTION,
+    combatAction: selectedMove
+  };
+  yield put(selectEnemyMoveAction);
+
+  const telegraphRoll = Math.random();
+  if (telegraphRoll > 0.7) {
+    yield sendMessage(
+      `${enemyCombatant.name} is about to ${selectedMove.name}`
+    );
+  } else {
+    yield sendMessage(`${enemyCombatant.name} disguises their attack`);
+  }
+}
+
+const calculateCounter = (
+  playerAction: CombatAction,
+  enemyAction: CombatAction
+): number => {
+  if (playerAction.type === enemyAction.type) {
+    return 0;
+  } else if (playerAction.speed === enemyAction.speed) {
+    if (playerAction.type === CombatActionType.Defense) {
+      return -2;
+    } else if (enemyAction.type === CombatActionType.Defense) {
+      return 2;
+    }
+  }
+  return 0;
+};
+
+function* dealDamage(damage: number, reciever: "Player" | "Enemy") {
+  const damageAction: AlterHealthAction | AlterEnemyHealthAction = {
+    type: reciever === "Player" ? ALTER_HEALTH : ALTER_ENEMY_HEALTH,
+    health: damage * -1
+  };
+
+  yield put(damageAction);
+}
+
+function* alterMomentum(momentum: number) {
+  const alterMomentumAction: AlterMomentumAction = {
+    type: ALTER_MOMENTUM,
+    momentum
+  };
+
+  yield put(alterMomentumAction);
+}
+
+function* defenseCheck(playerAction: CombatAction, enemyAction: CombatAction) {
+  if (playerAction.type === enemyAction.type) return;
+  if (playerAction.type === CombatActionType.Defense) {
+    yield alterMomentum(-1);
+    yield sendMessage(`Successful defense, gain 1 momentum`);
+  } else if (enemyAction.type === CombatActionType.Defense) {
+    yield alterMomentum(1);
+    yield sendMessage(`Successful enemy defense, lose 1 momentum`);
+  }
+}
+
+function* clashCheck(playerAction: CombatAction, enemyAction: CombatAction) {
+  if (
+    playerAction.type === CombatActionType.Offense &&
+    enemyAction.type === CombatActionType.Offense
+  ) {
+    yield dealDamage(playerAction.damage, "Enemy");
+    yield dealDamage(enemyAction.damage, "Player");
+    yield sendMessage(
+      `Both attacks land, you deal ${playerAction.damage} damage and take ${enemyAction.damage}`
+    );
+  }
+}
+
+function* sendMessage(message: string) {
+  const messageAction: AddMessageAction = {
+    type: ADD_MESSAGE,
+    message
+  };
+  yield put(messageAction);
+}
+
+export default function* rootSaga() {
+  yield all([combatActionListener(), enterCombatListener()]);
```

Comment:
I'd probably seperate these out into 2 different files... that will make them a bit more readable and easier to follow what's used where and what's common between the two sagas

Reply:
ah thats a good way of doing it, cheers